agent/engine/execcmd/manager_init_task_linux_test.go

```diff
@@ -194,3 +195,60 @@ func assertMountPoint(t *testing.T, c *apicontainer.Container, source, container
 	assert.Equal(t, readonly, found.ReadOnly, "Mount point for volume (%s) for container (%s) has the wrong write permissions", source, c.Name)
 	assert.Equal(t, containerPath, found.ContainerPath, "Mount point for volume (%s) points to the wrong container (%s) path", source, c.Name)
 }
+
+func TestGetExecAgentConfigFileName(t *testing.T) {
+	execAgentConfig := `
+	"Agent": {
+	  "Region": "",
+	  "OrchestrationRootDir": "",
+	  "ContainerMode": true,
+	  "Mgs": {
+		"Region": "",
+		"Endpoint": "",
+		"StopTimeoutMillis": 20000,
+		"SessionWorkersLimit": 2
+	  }
+	}`
+	sha := getExecAgentConfigHash(execAgentConfig)
+	configFileName := fmt.Sprintf("/execute-command/config/amazon-ssm-agent-%s.json", sha)
+	var tests = []struct {
+		fileExists             bool
+		createConfigFileErr    error
+		expectedConfigFileName string
+		expectedError          error
+	}{
+		{
+			fileExists:             true,
+			createConfigFileErr:    nil,
+			expectedConfigFileName: configFileName,
+			expectedError:          nil,
+		},
+		{
+			fileExists:             false,
+			createConfigFileErr:    nil,
+			expectedConfigFileName: configFileName,
+			expectedError:          nil,
+		},
+		{
+			fileExists:             false,
+			createConfigFileErr:    errors.New("cannot create config"),
+			expectedConfigFileName: "",
+			expectedError:          errors.New("cannot create config"),
+		},
+	}
+	for _, tc := range tests {
+		execAgentConfigFileExists = func(file string) bool {
+			return tc.fileExists
+		}
+		createNewExecAgentConfigFile = func(c, f string) error {
+			return tc.createConfigFileErr
+		}
+		defer func() {
```

Comment:
nit: you can put this `defer` outside the for (maybe just at the beginning of the function), this code creates 3 defers that are executed all after the overall test function is done